### PR TITLE
refactor: token balance error handling

### PR DESCRIPTION
**Pull request by magiziz**
# Description

Refactored token balance error message and ensured that token balances are only fetched again after 5 seconds if the token balance API fails.

## Type of change

- [x] Chore (non-breaking change that addresses non-functional tasks, maintenance, or code quality improvements)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Showcase (Optional)

image

# Checklist

- [x] Code in this PR is covered by automated tests (Unit tests, E2E tests)
- [x] My changes generate no new warnings
- [x] I have reviewed my own code
- [x] I have filled out all required sections
- [x] I have tested my changes on the preview link
- [ ] Approver of this PR confirms that the changes are tested on the preview link

### CODE REVIEW

File: apps/laboratory/tests/email.spec.ts [hunk 1]
```diff
@@ -100,9 +100,18 @@ emailTest('it should show names feature only for EVM networks', async ({ library
 emailTest('it should show loading on page refresh', async () => {
   await page.page.reload()
   await validator.expectConnectButtonLoading()
+  await page.page.waitForTimeout(5000)
```
Comment (tomiir) [inline apps/laboratory/tests/email.spec.ts:103]: is this required?

File: packages/core/src/controllers/AccountController.ts [hunk 2]
```diff
@@ -239,7 +248,10 @@ export const AccountController = {
         SwapController.setBalances(SwapApiUtil.mapBalancesToSwapTokens(response.balances))
       }
     } catch (error) {
-      SnackController.showError('Failed to fetch token balance')
+      if (chainId && state.tokenFetchFailureChainIds) {
+        state.tokenFetchFailureChainIds.add(chainId)
```
Comment (tomiir) [inline packages/core/src/controllers/AccountController.ts:252]: Are we not clearing this? What happens if there's a true flake in the request?
I think we should silently retry after some time

File: apps/laboratory/tests/email.spec.ts (hunk 1 above)
Comment (magiziz) [reply apps/laboratory/tests/email.spec.ts:103]: Yeah it'd need to finish loading before i can open the modal again

Comment (enesozturk) [reply apps/laboratory/tests/email.spec.ts:103]: But we don't know when the loading will finish right? Cannot be wait for an element to be visible instead?

Comment (tomiir) [reply apps/laboratory/tests/email.spec.ts:103]: I think best way here is to add a `validator.expectConnectButtonReady` or similar that does 
```
    const connectButton = this.page.getByTestId('connect-button')
    await expect(connectButton).toContainText('Connect', {
      timeout: MAX_WAIT
    }))
``` 
That way you do not use arbitrary timeouts

Comment (magiziz) [reply apps/laboratory/tests/email.spec.ts:103]: Good idea, let me add that 🙏

**Review by tomiir — APPROVED** [review]

**Review by enesozturk — APPROVED** [review]

**Pull request merged**